app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt

```diff
@@ -3,55 +3,163 @@ package com.jnj.vaccinetracker.register.dialogs
 import android.app.Dialog
 import android.os.Bundle
 import android.widget.Button
-import android.widget.CheckBox
 import android.widget.DatePicker
+import android.widget.LinearLayout
+import android.widget.NumberPicker
+import androidx.appcompat.widget.SwitchCompat
 import androidx.fragment.app.DialogFragment
 import com.jnj.vaccinetracker.R
-import com.jnj.vaccinetracker.common.helpers.findParent
 import com.soywiz.klock.DateTime
+import com.soywiz.klock.DateTimeSpan
 import java.util.Calendar
 
 class BirthDatePickerDialog(
-        private var selectedDate: DateTime? = null,
-        private var isBirthDateEstimatedChecked: Boolean = false)
-    : DialogFragment() {
+    private var selectedDate: DateTime? = null,
+    private var isBirthDateEstimatedChecked: Boolean = false,
+    private var yearsEstimated: Int? = null,
+    private var monthsEstimated: Int? = null,
+    private var daysEstimated: Int? = null
+) : DialogFragment() {
+
+    private lateinit var estimatedBirthdayLayout: LinearLayout
+    private lateinit var switchIsBirthDateEstimated: SwitchCompat
+    private lateinit var btnOk: Button
+    private lateinit var btnCancel: Button
+
+    private lateinit var datePicker: DatePicker
+    private lateinit var numberPickerYears: NumberPicker
+    private lateinit var numberPickerMonths: NumberPicker
+    private lateinit var numberPickerDays: NumberPicker
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
         val dialog = Dialog(requireContext())
         dialog.setContentView(R.layout.dialog_birth_date_picker)
 
-        val datePicker = dialog.findViewById<DatePicker>(R.id.datePicker)
-        val checkBox = dialog.findViewById<CheckBox>(R.id.checkbox_is_birth_date_estimated)
+        initializeViews(dialog)
+        setupDatePicker()
+        setupIsBirthDateEstimatedSwitch()
+        setupNumberPickers()
+        setupSwitchListener()
+
+        btnOk.setOnClickListener {
+            selectedDate = if (switchIsBirthDateEstimated.isChecked) {
+                calculateEstimatedDate()
+            } else {
+                DateTime(datePicker.year, datePicker.month + 1, datePicker.dayOfMonth)
+            }
+            isBirthDateEstimatedChecked = switchIsBirthDateEstimated.isChecked
```

Comment:
can we move it a little bit higher and use only this variable? it will avoid confusing between using sometimes "isBirthDateEstimatedChecked" and sometimes "switchIsBirthDateEstimated.isChecked". I mean something like:


btnOk.setOnClickListener {
            **isBirthDateEstimatedChecked = switchIsBirthDateEstimated.isChecked**
            selectedDate = if (isBirthDateEstimatedChecked) {
                calculateEstimatedDate()
            } else {
                DateTime(datePicker.year, datePicker.month + 1, datePicker.dayOfMonth)
            }
           val years = if (isBirthDateEstimatedChecked) numberPickerYears.value else null
            val months = if (isBirthDateEstimatedChecked) numberPickerMonths.value else null
            val days = if (isBirthDateEstimatedChecked) numberPickerDays.value else null
            (parentFragment as? BirthDatePickerListener)?.onBirthDatePicked(
                selectedDate!!,
                isBirthDateEstimatedChecked,
                years,
                months,
                days
            )
            dialog.dismiss()
        }

Reply:
done